edb/edgeql/compiler/viewgen.py

```diff
@@ -446,6 +473,133 @@ def gen_pointers_from_defaults(
     return [result[i] for i in topological.sort(graph, allow_unresolved=True)]
 
 
+def _apply_rewrites(
+    pointers: Dict[s_pointers.Pointer, Optional[irast.Set]],
+    specified_ptrs: Set[sn.UnqualName],
+    rewrite_kind: qltypes.RewriteKind,
+    view_scls: s_objtypes.ObjectType,
+    ir_set: irast.Set,
+    stype: s_objtypes.ObjectType,
+    path_id_namespace: Optional[irast.Namespace],
+    view_rptr: Optional[context.ViewRPtr],
+    exprtype: s_types.ExprType,
+    ctx: context.ContextLevel,
+) -> List[Tuple[s_pointers.Pointer, irast.Set | None]]:
+    result: List[Tuple[s_pointers.Pointer, irast.Set | None]] = []
+
+    path_id = ir_set.path_id
+    scls_pointers = stype.get_pointers(ctx.env.schema)
+
+    # init set that will be used for __subject__
+    subject_path = irast.PathId.from_type(
+        ctx.env.schema,
+        stype,
+        typename=sn.QualName(module="__derived__", name="__subject__"),
+    )
+    subject_set = irast.Set(
+        path_id=subject_path,
+        typeref=subject_path.target,
+        shape=tuple(
+            [
+                (ptr_set, qlast.ShapeOp.ASSIGN)
+                for ptr_set in pointers.values()
+                if ptr_set
+            ]
+        ),
+    )
+    ctx.env.set_types[subject_set] = stype
+
+    # init reference to std::bool
+    bool_type = ctx.env.schema.get("std::bool", type=s_types.Type)
+    bool_path = irast.PathId.from_type(
+        ctx.env.schema,
+        bool_type,
+        typename=sn.QualName(module="std", name="bool"),
+    )
+
+    # init set that will be used for __specified__
+    specified_path = irast.PathId.from_type(
+        ctx.env.schema,
+        stype,  # TODO: what's the type of this? named tuple?
```

Comment:
Also this.

Reply:
Yeah, you'll need to generate a fresh tuple type. There's a `create` method on the `Tuple` type in schema

Reply:
Oh, actually, since you're constructing the whole tuple here, try `setgen.new_tuple_set`